umm-spec-lib/src/cmr/umm_spec/migration/related_url_migration.clj

```diff
@@ -91,10 +91,24 @@
    Need to remove MimeType, add default for Name and Description if they don't exist."
   [element]
   (if-let [ol-resource (:OnlineResource element)]
-    (-> element
-        (assoc :OnlineResource {:Linkage (:Linkage ol-resource)
-                                :Name (util/with-default (:Name ol-resource) true)
-                                :Description (util/with-default (:Description ol-resource) true)}))
+    (assoc element :OnlineResource {:Linkage (:Linkage ol-resource)
+                                    :Name (util/with-default (:Name ol-resource) true)
+                                    :Description (util/with-default (:Description ol-resource) true)})
+    element))
```

Comment:
Does `assoc` not return what you were expecting here? It's inside the scope of the `if-let`, and it looks like you might not be getting the data that you're expecting

Reply:
If there is no OnlineResource in element, there is no point changing the OnlineResource(removing MimeType and add default values for the Name and Description), so just return element. 
Note: I'm not really changing the code here, but to refactor it so that we don't use threading when we have less than 3 things to thread.

Reply:
If you use assoc only, when there is no OnlineResource, it will return invalid OnlineResource that has nil Linkage(required field) and default Name and Description, which is wrong.